### DISCUSSION TITLE
Case sensitive identifiers

### DISCUSSION THREAD

**Discussion by harnash** [ory/kratos#2168] [Ideas]
I would like to as if a feature which allows to store and use case sensitive identifiers is something community would like to see in Kratos. This is mostly useful when dealing with legacy systems since I think current implementation (case insensitive) suites most modern use cases.

We are dealing with a legacy system where case sensitive login identifiers exists and this is causing issues when migrating user base to Kratos. We have implemented a feature (behind a flag) that enables Kratos to store case sensitive logins and we could share  it if that is something useful.

**Comment by zepatrik** [comment]
What if you would transform case-sensitive identifiers to case-insensitive ones? I.e. `Foo` becomes `+foo` while `a+b` becomes `a++b`? This would allow this feature without changing the inner logic. You only have to apply this mapping in the request handler.

**Reply by harnash** [reply]
But that would change how user needs to login no? We would like to avoid that for numerous reasons. We have been there :)

**Reply by harnash** [reply]
Are you suggesting adding internal only character that is not allowed as a part of the identifier and map it inside Kratos? That could work but will need quite a lot of work I guess since all APIs and user facing data would need transformations from/to this internal representation.
